Fix PODVector resize and reserve

AlexanderSinn opened the pull request:
Needed after https://github.com/AMReX-Codes/amrex/pull/4763

AlexanderSinn committed:
Fix PODVector resize and reserve

AlexanderSinn committed:
remove const

AlexanderSinn committed:
Update ParticleTile and SoA

ax3l commented on src/Base/PODVector.cpp:
```diff
@@ -72,9 +72,12 @@ void make_PODVector(py::module &m, std::string typestr, std::string allocstr)
+        .def("resize", [](PODVector_type & pv, std::size_t new_size, const T& init_val){
+            pv.resize(new_size, init_val); })
+        .def("reserve", [](PODVector_type & pv, std::size_t new_capacity){
+            pv.reserve(new_capacity); })
```
Thank you!

Let's improve this a bit to add:
- enums of the new arguments
- expose the new signatures with
- named arguments with
- default values

Please let me know if you need help.

AlexanderSinn replied:
Is there a way to expose enum struct to python or should it just be a string?

WeiqunZhang replied:
I did something like this in another place. It seems to work.
```
// C++
    py::enum_<EBSupport>(m, "EBSupport")
        .value("basic", EBSupport::basic)
        .value("volume", EBSupport::volume)
        .value("full", EBSupport::full);
```

```
# python
factory = amr.makeEBFabFactory(geom, ba, dm, ng, amr.EBSupport.full)
```

ax3l replied:
Yep, you can use the new Enum support that I use in #454 as well, pybind11 recently modernized it. Both either works.

ax3l committed:
PODVector: New Overloads & Defaults

ax3l committed:
AMReX: `development`

ax3l committed:
Fix ParmParse: Enum

pybind11 cannot use `overload_cast` anymore if one of
the overloads has an enable_if template and the others
are not templated.

Express as lambdas.

ax3l committed:
Two More Resizes

ax3l committed:
Native_Enum

pybind11-stubgen needs location hint for default arguments
that are enums, even if defined early enough.

ax3l committed:
`enum.Enum`

ax3l committed:
Stubs: 1d/2d/3d
